include/cantera/zeroD/FlowDevice.h

```diff
@@ -131,10 +131,12 @@ class FlowDevice
 
     //! Set the fixed mass flow rate (kg/s) through the flow device.
     //! @deprecated To be removed after Cantera 2.5.
-    void setMassFlowRate(double mdot) {
-        warn_deprecated("FlowDevice::setMassFlowRate()",
+    //! @todo make MassFlowController::setMassFlowRate non-virtual after 2.5
+    virtual void setMassFlowRate(double mdot) {
```

Comment:
I don't think this method should be `virtual`.

With this method marked `virtual`, use of `setMassFlowRate` on a `MassFlowController` accessed as a `FlowDevice*` will succeed without a deprecation warning, but then become an error when we remove the base class method.

If this method is non-virtual, use of `setMassFlowRate` on a `MassFlowController` accessed as a `FlowDevice*` will raise a deprecation warning which then becomes an error when we remove the base class method.

Reply:
Good catch! (And easily fixed)